test/test_nn.py

```diff
@@ -2983,14 +2983,14 @@ def test_data_parallel_nested_output(self):
         def fn(input):
             return [
                 input, (input.sin(), input.cos(), [input.add(1)]), input,
-                {'a': input, 'b': [input.sin()]}
```

Comment:
Would it be better to change self.assertEquals to do better dict equality checks?

Reply:
Yep we should do that too. Although we should still apply this fix because it is a stricter check that we maintain the ordering in DP.

Reply:
#11094